lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -34,8 +35,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
     }
 
     // Use custom serialize if it exists.
-    if (handler.serialize) {
-      return handler.serialize(model, names);
+    if (serializer) {
+      return serializer(model, names);
```

Comment:
I _really_ appreciate the thought you've put into this. We'll discuss this more and come to a conclusion. In fact you can follow along in [#dev-router](https://embercommunity.slack.com/messages/dev-router/) on the [Ember Community Slack](https://ember-community-slackin.herokuapp.com/) if you wish. 😄